ui/src/app/subnets/views/VLANDetails/VLANSummary/VLANSummary.tsx

```diff
@@ -1,89 +1,77 @@
+import authSelectors from "app/store/auth/selectors";
 import type { RootState } from "app/store/root/types";
 import vlanSelectors from "app/store/vlan/selectors";
 import type { VLAN, VLANMeta } from "app/store/vlan/types";
-import { breakLines } from "app/utils";
 
 type Props = {
   id: VLAN[VLANMeta.PK] | null;
 };
 
-const getRackIDs = (vlan: VLAN | null) => {
-  const rackIDs = [];
-  if (vlan) {
-    if (vlan.primary_rack) {
-      rackIDs.push(vlan.primary_rack);
-    }
-    if (vlan.secondary_rack) {
-      rackIDs.push(vlan.secondary_rack);
-    }
-  }
-  return rackIDs;
-};
-
 const VLANSummary = ({ id }: Props): JSX.Element | null => {
+  const isAdmin = useSelector(authSelectors.isAdmin);
   const vlan = useSelector((state: RootState) =>
     vlanSelectors.getById(state, id)
   );
-  const controllers = useSelector((state: RootState) =>
-    controllerSelectors.getByIDs(state, getRackIDs(vlan))
-  );
-  const controllersLoading = useSelector(controllerSelectors.loading);
+  const [editing, setEditing] = useState(false);
+  const canEdit = isAdmin;
 
-  if (!vlan) {
+  if (!id || !vlan) {
     return null;
   }
   return (
-    <TitledSection title="VLAN summary">
-      <Row>
-        <Col size={6}>
-          <Definition label="VID" description={`${vlan.vid}`} />
-          <Definition label="Name" description={vlan.name} />
-          <Definition label="MTU" description={`${vlan.mtu}`} />
-          <Definition label="Description" description={vlan.description} />
-        </Col>
-        <Col size={6}>
-          <Definition label="Space">
-            <SpaceLink id={vlan.space} />
-          </Definition>
-          <Definition label="Fabric">
-            <FabricLink id={vlan.fabric} />
-          </Definition>
-          <Definition
-            label={
-              <>
-                Rack controllers
-                <Tooltip
-                  className="u-nudge-right--small"
-                  message={breakLines(
-                    "A rack controller controls hosts and images and runs network services like DHCP for connected VLANs."
-                  )}
-                >
-                  <Icon name="information" />
-                </Tooltip>
-              </>
-            }
+    <TitledSection
+      title="VLAN summary"
+      buttons={
+        canEdit &&
+        !editing && (
+          <Button
+            appearance="neutral"
+            className="u-no-margin--bottom"
+            onClick={() => setEditing(true)}
           >
-            {controllersLoading ? (
-              <Spinner />
-            ) : (
-              controllers.map((controller) =>
-                controller ? (
-                  <ControllerLink key={controller.system_id} {...controller} />
-                ) : null
-              )
-            )}
-          </Definition>
-        </Col>
-      </Row>
+            Edit
+          </Button>
+        )
+      }
+    >
+      {editing ? (
+        <EditVLAN
```

Comment:
You may want to wrap the form in `<Col size={6}>` as well to better match the  width of non-editable description:
https://github.com/canonical-web-and-design/maas-ui/pull/3579/files#diff-2b200479e25c140c7a2e065060a6106e22e33f26d8d6477de969867790a1df94R57-R73

Reply:
The form fields are inside cols in the form here: https://github.com/canonical-web-and-design/maas-ui/pull/3579/files#diff-d81a78842ab4a9e1d906dd2f0c17674ec821a44ee128f2a2506f9c982785d745R97-R111 or am I missing something?